### PR TITLE
reuse configuration from parent deck file

### PR DESCRIPTION
**Pull request by tvidal-net**
After asking #81 I was looking at the code and decided to make an attempt of an implementation.
Here's the PR. I have never written go code before, please excuse any gross mistake.

Any feedback is welcome. If you think there is value in this PR, I can update the configuration example to reflect this change.

### CODE REVIEW

**Comment by muesli** [comment]
> After asking #81 I was looking at the code and decided to make an attempt of an implementation. Here's the PR. I have never written go code before, please excuse any gross mistake.

Well done for a first attempt at Go!
 
> Any feedback is welcome. If you think there is value in this PR, I can update the configuration example to reflect this change.

Since this is a breaking change in behavior, I think the big question is how we can support optionally without breaking existing configurations. I can absolutely see a use-case for this, but it would also break a couple of my decks currently.

**Comment by tvidal-net** [comment]
Thank you for your response.

I don't think I understand what you mean by "breaking change in behaviour". Deck files that do not have the `parent = "parent.deck"` tag will work exactly as they did before.

Whenever a deck config has the `parent = "parent.deck"` tag, then the `parent.deck` config is loaded recursively and overridden by the current config keys and background.

You mentioned this would also break a couple of your decks currently, can you please give an example?

Thank you! Best regards!

**Comment by muesli** [comment]
> I don't think I understand what you mean by "breaking change in behaviour". Deck files that do not have the `parent = "parent.deck"` tag will work exactly as they did before.
> 
> Whenever a deck config has the `parent = "parent.deck"` tag, then the `parent.deck` config is loaded recursively and overridden by the current config keys and background.

This is the embarrassing moment where I'll have to admit that I apparently failed to fully understand your change during my first review :smile: For some reason I had assumed the child-decks would automatically inherit their parent.

You're right then, this change makes sense and is probably the only way we can ensure to support existing deck configurations.

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,7 @@ dist/
 
 # direnv
 .envrc
+
+# GoLand Project Files
```
Comment (muesli) [inline .gitignore:23]: Must admit I'm not a fan of pushing these ignores to individual repos. That's probably something that should go in your user's (or system's) global gitignore config. If you don't mind I'll drop this commit from the PR.

**Review by muesli — APPROVED** [review]

**Comment by muesli** [comment]
Thank you @tvidal-net!

**Pull request merged**